### PR TITLE
Updating the architecture diagram

### LINKED ISSUE #307: Bug: The architecture diagram is not updated
### Where's the bug?

https://www.risingwave.dev/docs/latest/architecture/

### Describe the bug

The diagram on the Architecture page should reflect the latest changes like Sink and other new components.

### PR DESCRIPTION
**Pull request by CharlieSYH**
## Info

Resolves https://github.com/risingwavelabs/risingwave-docs/issues/307

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/Architecture.md [hunk 1]
```diff
@@ -11,11 +11,10 @@ RisingWave includes these key components:
 
 * A serving layer that parses SQL queries and performs planning and optimizations of query jobs.
 * A processing layer that performs all the data computation and status updates.
-* A meta server that coordinates the operations between the serving layer and the processing layer.
-* A persistence layer that stores data to and retrieves data from object storage like S3.
-
+* A metadata management layer that coordinates the operations across all other layers.
```
Comment (hengm3467) [inline docs/Architecture.md]: A metadata management service that manages the metadata of different nodes, and coordinates operations among these nodes. It consists of a local metadata store and various management modules.

**Review by lmatz — APPROVED** [review]

File: docs/images/architecture.png
Comment (fuyufjh) [inline docs/images/architecture.png:1]: Prefer to omit the "Connector Node" because it should be deployed as sidecar in the future.

Comment (CharlieSYH) [reply docs/images/architecture.png:1]: @fuyufjh Which is better? Connector node in dashes or no connector node.
image
image

Comment (fuyufjh) [reply docs/images/architecture.png:1]: Prefer the later one: no connector node

**Review by hengm3467 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**